async/abortable.ts

```diff
@@ -77,11 +125,9 @@ function abortablePromise<T>(
   p: Promise<T>,
   signal: AbortSignal,
 ): Promise<T> {
-  if (signal.aborted) {
-    return Promise.reject(createAbortError(signal.reason));
-  }
+  if (signal.aborted) return Promise.reject(signal.reason);
```

Comment:
On 2nd thought. Maybe this should throw. Thoughts?

Reply:
This looks fine to me